.github/workflows/package.yml

```diff
@@ -8,8 +13,69 @@ on:
   workflow_dispatch:
   repository_dispatch:
 jobs:
+  build-test-tools:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Cache test tools
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.TEST_TOOLS_DIR }}
+          key: ${{ runner.os }}-test-tools-${{ github.workflow_sha }}
+      - name: Set up Go
+        if: steps.cache.outputs.cache-hit != 'true'
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
+      - name: Build Emutest binary
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          git clone https://github.com/sehugg/emutest
+          cd emutest
+          go install
+          mkdir -p ${{ env.EMUTEST_DIR }}
+          GOPATH=$(go env GOPATH)
+          cp $GOPATH/bin/emutest ${{ env.EMUTEST_DIR }}
+      - name: Fetch Mesen
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p ${{ env.MESEN_DIR }}
+          cd ${{ env.MESEN_DIR }}
+          #curl -Lo Mesen.exe https://github.com/NovaSquirrel/Mesen-X/releases/download/1.0.2/Mesen.exe
+          curl -Lo mesen.zip https://github.com/NovaSquirrel/Mesen-X/releases/download/1.0.1/Mesen-Linux.2022-05-14.zip
+          unzip mesen.zip
+          chmod +x Mesen.exe
+          echo '#!/bin/sh' > mesen
+          echo "xvfb-run -a /usr/bin/mono $(pwd)/Mesen.exe \"\$@\"" >> mesen
+          chmod +x mesen
+      - name: Build Stella2014 core
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          git clone https://github.com/libretro/stella2014-libretro
+          cd stella2014-libretro
+          make --silent -j 4
+          mkdir -p ${{ env.LIBRETRO_CORES_DIR }}
+          mv stella2014_libretro.so ${{ env.LIBRETRO_CORES_DIR }}
+      - name: Build Nestopia core
```

Comment:
Does the mesen libretro core not work? If so, do we have any insight into why not? It's not exactly desirable to require three separate NES emulators to test the SDK; it would be more maintainable to use just native Mesen if we can't avoid it.

Reply:
I did get the Mesen-X Libretro core working, and was surprised to see it failed on exactly the same two tests as FCEUMM, chr-rom and chr-swap-split. I think maybe the problem is that the GTROM 111 mapper does not support CHR ROM. @asiekierka do you have any more info on this?

Assuming we figure this out, all the tests will pass with both Mesen-X and FCEUMM cores, and we can lose native Mesen and Nestopia. I think it's good to have two NES emulators in case a test author wants to compare results.

Reply:
There is no "mapper which doesn't support CHR-ROM". Nothing prevents you from wiring ROM in place of RAM in this regard; it's not like PRG-ROM and PRG-RAM. If an emulator doesn't emulate that combination, it's the fault of the emulator.